### PR TITLE
Scipy toolkit

### PR DESCRIPTION
**Pull request by pkohvaei**
Sparse matrix manipulation tool added. Updates.

### CODE REVIEW

**Comment by bgruening** [comment]
Hi,

this one file is really big. Can you shorten it or use `contains` as test option from Galaxy?

**Comment by bgruening** [comment]
Also try to include such a .shed.yaml file to trigger automatic tests :)

**Comment by pkohvaei** [comment]
Which file?

**Comment by gregvonkuster** [comment]
@pkohvaei This one is the likely candidate: https://github.com/pkohvaei/galaxytools/blob/scipy-toolkit/tools/sklearn_tools/test-data/sparse.mtx

File: tools/sklearn_tools/test-data/linear_model01.txt [hunk 1]
```diff
@@ -1,184 +0,0 @@
-ccopy_reg
-_reconstructor
```
Comment (bgruening) [inline tools/sklearn_tools/test-data/linear_model01.txt]: this file is used in the discriminant tool, so you can not simply remove it ...

File: tools/sklearn_tools/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  Scikit-learn is an open source machine learning library written in python.
+  It provides different flavors of machine learning algorithms for classification,
+  regression, and clustering. It has designed to interoperate with python numerical
+  and scientific libraries Numpy and Scipy.
+  
+  The official repository of Scikit-learn is at https://github.com/scikit-learn/scikit-learn
+
+categories:
+  - Machine Learning
+  - Statistical Pattern Recognition
```
Comment (bgruening) [inline tools/sklearn_tools/.shed.yml]: These needs to correspond to the TS categories: https://testtoolshed.g2.bx.psu.edu/
`Statistics` would be fine.

File: tools/sklearn_tools/sample_generator.xml [hunk 3]
```diff
@@ -231,87 +230,73 @@ targets_df.to_csv(path_or_buf = "$outfile_targets", sep="\t", index=False, heade
...
-            <output name="outfile_points" file="gaus_points.txt"/>
-            <output name="outfile_targets" file="gaus_targets.txt"/>
+            <output name="outfile" file="gaus.txt"/>
         </test>
         <test>
             <param name="selected_generator" value="make_hastie_10_2"/>
             <param name="random_state" value="100"/>
-            <output name="outfile_points" file="hastie_points.txt"/>
-            <output name="outfile_targets" file="hastie_targets.txt"/>
+            <output name="outfile" file="hastie.txt"/>
```
Comment (bgruening) [inline tools/sklearn_tools/sample_generator.xml]: This file is too large. `contains` can be used.

File: tools/sklearn_tools/.shed.yml (hunk 2 above)
Comment (pkohvaei) [inline tools/sklearn_tools/.shed.yml]: Done.

File: tools/sklearn_tools/sample_generator.xml (hunk 3 above)
Comment (pkohvaei) [inline tools/sklearn_tools/sample_generator.xml]: Done.

**Comment by bgruening** [comment]
Let's see if the tests are passing.

**Comment by bgruening** [comment]
One test still failing :-(

**Comment by pkohvaei** [comment]
I think now the tests would pass.

**Comment by pkohvaei** [comment]
I try to avoid adding sim_size just to see if the computed model is really reproducible. Thus now one time control parameter is deleted from the test.
Could you test again?

File: tools/sklearn_tools/eden_converter.xml [hunk 4]
```diff
@@ -4,6 +4,8 @@
         <import>main_macros.xml</import>
     </macros>
     <expand macro="python_requirements"/>
+    <requirement type="package" version="2.2.5">viennarna</requirement>
```
Comment (bgruening) [inline tools/sklearn_tools/eden_converter.xml]: This can not work. `requirement` needs to be in a `requirements` sections.

**Comment by bgruening** [comment]
Great, tests seems to pass. I rebased your branch. Let's see if it is still passing!

**Pull request merged**